### PR TITLE
Add compile jack comand to cli

### PR DESCRIPTION
**Pull request by happytomatoe**
As I am not using web-ide jack editor because it's slow, I've decided to add cli command to invoke Jack compiler directly

Screenshot from 2024-09-16 17-03-14

### CODE REVIEW

**Comment by DavidSouther** [comment]
@happytomatoe thank you so much for this PR! Definitely something we've had as a wishlist item while our effort has been focused on the web IDE. For your purposes, you might also look into a "CLI runner", that provides a CLI or (lightweight) windowed environment for Screen + Keyboard. I'll have a review for you later today. It looks like you can start by running `npm run ci` locally to ensure your CI will pass, and `npm run format ; npm run fix` to apply automatic formatting and lint fixes.

File: cli/src/index.ts [hunk 1]
```diff
@@ -74,6 +78,96 @@ yargs(hideBin(process.argv))
...
+          await fs.writeFile(outpath, compiled);
+        } else {
+          if (!error) {
+            console.error("Compilation failed\n");
+          }
+          console.error(
+            Colors.Red +
+              compiled.message.replace(
+                /Line\s([\d]+):/g,
+                name + ".jack:$1" + Colors.Reset,
```
Comment (happytomatoe) [inline cli/src/index.ts:158]: FYI - I am using vscode and if  the message in this format one can ctr+click to go that line

**Comment by happytomatoe** [comment]
@DavidSouther Could you elaborate on CLI runner? Currently in` cli/src/index.ts `I see that run command only accepts` .tst `files

File: cli/src/index.ts (hunk 1 above)
Comment (DavidSouther) [reply cli/src/index.ts:158]: Nice!

**Comment by DavidSouther** [comment]
>  I see that run command only accepts`.tst`files

Yes; this would be an entirely new project. `nand2tetris run file.tst` would still run a .tst file as expected. This new mode might be `nand2tetris exec myprog.hack`, where `myprog.hack` is the final compiled .jack -> vm -> hack binary file. This mode would then have the screen and keyboard interaction. Maybe we should start a new thread to discuss that!

**Comment by DavidSouther** [comment]
Thank  you @happytomatoe for your first PR!

**Pull request merged**